components/common/Button.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { ReactNode } from 'react';
+
+type Variant = 'primary' | 'secondary' | 'modal' | 'inactive';
+type Size = 'desktop' | 'tablet' | 'mobile' | 'large' | 'small' | 'sign';
+
+interface ButtonProps {
+  variant: Variant;
+  type?: 'submit' | undefined;
+  size: Size;
+  className?: String;
+  onClick?: () => void;
+  children: ReactNode;
+}
+
+function Button({
+  variant,
+  type,
+  size,
+  className,
+  onClick,
+  children,
+}: ButtonProps) {
+  let combinedClassName = '';
+  switch (variant) {
+    case 'primary': {
+      combinedClassName =
+        'border-none rounded-md flex-center font-medium text-white bg-violet';
+      break;
+    }
+    case 'secondary': {
+      combinedClassName =
+        'border border-gray30 rounded-md flex-center font-medium text-violet bg-white';
+      break;
+    }
+    case 'modal': {
+      combinedClassName =
+        'border border-gray30 rounded-md flex-center font-medium text-gray50 bg-white';
+      break;
+    }
+    case 'inactive': {
+      combinedClassName =
+        'border-none rounded-md flex-center font-medium text-white bg-gray40';
+      break;
+    }
+  }
+
+  switch (size) {
+    case 'desktop': {
+      combinedClassName += ' py-[0.4375rem] px-[1.8125rem] text-sm ';
+      break;
+    }
+    case 'tablet': {
+      combinedClassName += ' py-[0.375rem] px-[1.4375rem] text-sm ';
+      break;
+    }
+    case 'mobile': {
+      combinedClassName += ' py-[0.4375rem] px-[2.3125rem] text-xs ';
+      break;
+    }
```

Comment:
스크린 따라 달라지는 것을 아예 반응형으로 구현하면 어떨까요?
```suggestion
    case 'medium': {
      combinedClassName += ' desktop:py-[0.4375rem] desktop:px-[1.8125rem] tablet:py-[0.375rem] tablet:px-[1.4375rem] tablet:text-sm mobile:py-[0.4375rem] mobile:px-[2.3125rem] mobile:text-xs';
      break;
    }